### ISSUE TITLE
k3s-agent attempts to add iptables rule with empty --dports and fails to start

### ISSUE THREAD

**Issue by discordianfish** [k3s-io/k3s#2998]
**Environmental Info:**
K3s Version:
k3s version v1.20.4+k3s1 (838a906a)

Node(s) CPU architecture, OS, and Version: 
Linux filer 4.19.0-14-amd64 #1 SMP Debian 4.19.171-2 (2021-01-30) x86_64 GNU/Linux

**Describe the bug:**
Starting k3s-agent fails with this error:
```
Feb 28 10:01:14 filer k3s[19469]: I0228 10:01:14.751191   19469 network_policy_controller.go:138] Starting network policy controller
Feb 28 10:01:14 filer k3s[19469]: F0228 10:01:14.770233   19469 network_policy_controller.go:261] Failed to verify rule exists in KUBE-ROUTER-INPUT chain due to running [/usr/sbin/iptables -t filter -C KUBE-ROUTER-INPUT -p tcp -m comment --comment allow LOCAL TCP traffic to node ports - NTGHJFAJAOEPWU6M -m addrtype --dst-type LOCAL -m multiport --dports  -j RETURN --wait]: exit status 2: iptables v1.8.2 (legacy): invalid port/service `' specified
Feb 28 10:01:14 filer k3s[19469]: Try `iptables -h' or 'iptables --help' for more information.
Feb 28 10:01:14 filer systemd[1]: k3s-agent.service: Main process exited, code=exited, status=255/EXCEPTION
Feb 28 10:01:14 filer systemd[1]: k3s-agent.service: Failed with result 'exit-code'.
```

Apparently it tries to use --dports without specifying the ports.

**Steps To Reproduce:**
- fresh install on debian buster
- linked iptables to iptables-legacy as described in https://rancher.com/docs/k3s/latest/en/advanced/#enabling-legacy-iptables-on-raspbian-buster (though its not raspbian but vanilla buster also uses nft
- did not do any customization, so just:
```
curl -sfL https://get.k3s.io | K3S_TOKEN=xxx K3S_URL=https://... sh -
```

**Comment by discordianfish** [comment]
Looks like this is empty in my case: https://github.com/k3s-io/k3s/blob/65c78cc39790b46b84813f1d6cdaba13132384b6/pkg/agent/netpol/network_policy_controller.go#L325 - I didn't specify --service-node-port-range though.

Any chance that #2867 introduced a dependency on the control node providing the ports, which it doesn't in 1.20.2 I'm running?

**Comment by discordianfish** [comment]
Looks like that's the case. Updating master to 1.20.4 fixed the issue. But I'd argue there should be a fix to make #2867 backwards compatible or come up with a different versioning scheme to avoid surprising user with such breakage in a minor version update.

**Comment by brandond** [comment]
Duplicate of https://github.com/k3s-io/k3s/issues/2996#issuecomment-787442611
